### PR TITLE
Add support for UECM Registration and Deregistration for a PDU Session

### PR DESCRIPTION
**Pull request by saileshvvr**
Added support in SMF for UECM Registration and Deregistration for a PDU Session

### CODE REVIEW

File: internal/sbi/consumer/ue_context_management.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		smCtx.Supi, smCtx.PduSessionId)
+	defer func() {
+		if httpResp != nil {
+			if rspCloseErr := httpResp.Body.Close(); rspCloseErr != nil {
+				logger.ConsumerLog.Errorf("UeCmDeregistration response body cannot close: %+v",
+					rspCloseErr)
+			}
+		}
+	}()
+	if localErr == nil {
```
Comment (ianchen0119) [inline internal/sbi/consumer/ue_context_management.go:87]:
```go
smCtx.UeCmRegistered = false
```

File: internal/sbi/producer/pdu_session.go [hunk 2]
```diff
@@ -314,6 +323,20 @@ func HandlePDUSessionSMContextUpdate(smContextRef string, body models.UpdateSmCo
...
+				if problemDetails != nil {
+					if problemDetails.Cause != "CONTEXT_NOT_FOUND" {
+						logger.PduSessLog.Errorf("UECM_DeRegistration Failed Problem[%+v]", problemDetails)
+					}
+				} else if err != nil {
+					logger.PduSessLog.Errorf("UECM_DeRegistration Error[%+v]", err)
+				} else {
+					logger.PduSessLog.Traceln("UECM_DeRegistration successful")
+				}
+				smContext.UeCmRegistered = false
```
Comment (ianchen0119) [inline internal/sbi/producer/pdu_session.go:337]: should change the state in `UeCmDeregistration()`

**Review by ianchen0119 — CHANGES_REQUESTED** [review]

**Comment by ianchen0119** [comment]
Hi @saileshvvr 

Thanks for your PRs.
Please help to fix the ci-lint error when you feel free.

File: internal/sbi/consumer/ue_context_management.go (hunk 1 above)
Comment (saileshvvr) [reply internal/sbi/consumer/ue_context_management.go:90]: Thanks, As suggested, updated the state in the UeCmDeregistration()

File: internal/sbi/producer/pdu_session.go (hunk 2 above)
Comment (saileshvvr) [reply internal/sbi/producer/pdu_session.go]: Thanks. Removed the code duplication and set the state inside UeCmDeregistration()

**Review by ianchen0119 — APPROVED** [review]

**Comment by saileshvvr** [comment]
Hi @ianchen0119 
Thanks for adding the commit to resolve ci linter.

**Comment by saileshvvr** [comment]
Hi @ianchen0119 
Could you please help share process for merging the PR.

**Comment by ianchen0119** [comment]
Hi @saileshvvr 

This PR will also be reviewed by @tim-ywliu 
It will be merged once it gets the approval by him.

Thank you.

**Comment by saileshvvr** [comment]
Thanks @ianchen0119

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**